torch/lib/c10d/TCPStore.cpp

```diff
@@ -57,10 +56,165 @@ TCPStoreDaemon::~TCPStoreDaemon() {
   closeStopSignal();
 }
 
-void TCPStoreDaemon::join() {
+void BackgroundThread::join() {
   daemonThread_.join();
 }
 
+#ifdef _WIN32
```

Comment:
(can be done in follow up PRs) to get rid of frequent `#ifdef`, we can have two implementations for Unix and Windows respectively. `UnixSockUtils.hpp` and `WinSockUtils.hpp` can serve as an example.